rpc/RPCSession.java

```diff
@@ -45,18 +47,22 @@
     private final GraknGrpc.GraknBlockingStub blockingGrpcStub;
 
     public RPCSession(GraknClient client, String database, Type type, GraknOptions options) {
-        this.channel = client.channel();
-        this.database = database;
-        this.type = type;
-        blockingGrpcStub = GraknGrpc.newBlockingStub(channel);
-
-        final SessionProto.Session.Open.Req openReq = SessionProto.Session.Open.Req.newBuilder()
-                .setDatabase(database).setType(sessionType(type)).setOptions(options(options)).build();
-
-        sessionId = blockingGrpcStub.sessionOpen(openReq).getSessionId();
-        pulse = new Timer();
-        isOpen = new AtomicBoolean(true);
-        pulse.scheduleAtFixedRate(this.new PulseTask(), 0, 5000);
+        try {
+            this.channel = client.channel();
+            this.database = database;
+            this.type = type;
+            blockingGrpcStub = GraknGrpc.newBlockingStub(channel);
+
+            final SessionProto.Session.Open.Req openReq = SessionProto.Session.Open.Req.newBuilder()
+                    .setDatabase(database).setType(sessionType(type)).setOptions(options(options)).build();
+
+            sessionId = blockingGrpcStub.sessionOpen(openReq).getSessionId();
+            pulse = new Timer();
+            isOpen = new AtomicBoolean(true);
+            pulse.scheduleAtFixedRate(this.new PulseTask(), 0, 5000);
+        } catch (StatusRuntimeException e) {
+            throw new GraknClientException(e);
+        }
```

Comment:
It's worth noting that we should do the same change in `close` - however, I've already done so in this PR: https://github.com/graknlabs/client-java/pull/227; so don't worry about it.